### ISSUE TITLE
[RFE] Provide a SCM update option to add `--force` flag to all role updates.

### ISSUE THREAD

**Issue by benjdudas** [ansible/awx#4606]
##### ISSUE TYPE
 - Feature Idea
 
##### SUMMARY

Forced role overwrite is currently dependent on the revision of the parent project. Updating the roles without modifying the parent project can result in stale roles being used for the job run. Setting project options "update on launch + delete on update" is a workaround for a single node install, but when running as a cluster only the execution node selected for the scm_update will install the updated roles within the cache timeout. This can cause an issue when running the jobs concurrently across multiple nodes. The other workaround here would be to update the requirements.yml file in the project with version info of the role so that there is a revision change on the project.

Since it is not ideal to have a forced overwrite of roles in all use cases, it would be cool if we had the option per project to always use the `fetch galaxy roles from requirements.yml (forced update)` task for project updates. This would probably require a modification of the conditionals for the role update tasks in /awx/playbooks/project_update.yml and the introduction of a new variable (e.g. scm_force).

**Comment by ryanpetrello** [comment]
@AlanCoding you have any thoughts on this in the context of the new branching work?

**Comment by AlanCoding** [comment]
For 3.6 we are not supporting use of cached roles or collections at all. Although we are not literally using the `--force` flag, we are doing the same thing in effect, because we start with a clean install directory.

We tabled the discussion about a caching mechanism for collections and roles (and got rid of the old way). Firstly, we need a concrete idea of how versioning of collections works (we're getting there), and how it interacts with an implementation of collections as a top-level item in Tower. This is what we discussed for 3.6, but then punted. It remains a highly relevant topic for 3.7 planning.

tl:dr;

 - valid for 3.5.x
 - should not apply for 3.6
 - @benjdudas has several good points to consider in 3.7

**Comment by AlanCoding** [comment]
Roles are not cached anymore. We do not use the `--force` flag, but install collections & roles to a clean directory. So this should not be relevant in AWX anymore.

**Comment by wenottingham** [comment]
Closing, then.
